### ISSUE TITLE
Parameter for @classmethod marked as missing

### ISSUE THREAD

**Issue by Ampfing** [microsoft/pyright#5079]
<h3> Environment data </h3>

-   Language Server version: 2023.5.10
-   OS and version: win32 x64
-   Python version (and distribution if applicable, e.g. Anaconda): 3.9.12 64-bit
-   python.analysis.indexing: true
-   python.analysis.typeCheckingMode: basic

<h3> Code Snippet </h3>

```python
from abc import ABCMeta

from typing import Dict, Any

class Singleton(ABCMeta):
    __instances: Dict[Any, Any] = {}
    def __call__(cls, *args: Any, **kwargs: Any) -> Any:
        if cls not in cls.__instances:
            cls.__instances[cls] = super(Singleton, cls).__call__(*args, **kwargs)
        return cls.__instances[cls]

    @classmethod
    def delete_all_instances(cls) -> None:
        cls.__instances = {}
        print("Deleted")

class Class1(metaclass=Singleton):
    def __init__(self) -> None:
        print("Constructor")
        self.__name = 'Class1'

    def say_name(self) -> str:
        return self.__name


if __name__ == "__main__":
    obj1 = Class1()
    print(obj1.say_name())
    Singleton.delete_all_instances()
    obj1 = Class1()
    print(obj1.say_name())

```

<h3> Repro Steps </h3>

Save a python file with above code so Pylance runs

<h3> Expected behavior </h3>

No error should be displayed. Code runs perfectly fine during exection

<h3> Actual behavior </h3>

Line 29 (`Singleton.delete_all_instances()`) is marked with `Argument missing for parameter "cls"`

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release of pyright.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.307, which I just published. It will also be included in a future release of pylance.

**Comment by ogenstad** [comment]
Did this ever get fixed in pylance? Still seeing the issue with version v2023.9.10.

**Comment by erictraut** [comment]
Yes, this issue was fixed. If you think that you are seeing a similar issue, please file a new bug report with a code sample and repro steps.

**Comment by ogenstad** [comment]
Ok, thanks. Was using the file above as a test in one of my repos. Can now see that it's working if I run it stand alone in another virtualenv. So I probably have some package that breaks it, will try to track it down.
